File: Services/CommonLib/src/main/java/nuricanozturk/dev/dto/CreatePrescriptionDTO.java
package nuricanozturk.dev.dto;

import java.io.Serializable;
import java.util.List;

    public record CreatePrescriptionDTO(String pharmacyName, String pharmacyUsername, String customerIdentityNumber, String customerFullName,
                                    List<MedicineDTO> medicines, double totalPrice) implements Serializable
{
}

File: Services/medicineApp/src/app/dto/CreatePrescriptionDTO.ts
import {MedicineDTO} from "./MedicineDTO";
import {getUser} from "../services/login.service";

export class CreatePrescriptionDTO {
  public pharmacyName: string;
  public pharmacyUsername: string;
  public customerIdentityNumber: string;
  public customerFullName: string;
  public totalPrice: number;
  public medicines: MedicineDTO[]

  constructor() {
    const user = getUser()
    this.pharmacyName = user.pharmacyName
    this.pharmacyUsername = user.username
  }
}
